### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by libuweber**
# Pull Request Template

## Description

Adds unit testing infrastructure and a few basic tests to start with.

## How has this been tested?
Unit Tests are integrated in the ci workflow.

**Versions**
* Microgateway: 2.0
* Helm Chart: 0.7.0

## Checklist:
- [ x] The code has been reviewed (self-review, ...).
- [ ] The parts of the code which are hard to understand are commented.
- [ ] The corresponding documentation has been updated.
- [x] The changes do not cause warnings.
- [ ] The spelling has been checked.

### CODE REVIEW

File: charts/microgateway/tests/__snapshot__/configmap_test.yaml.snap [hunk 1]
```diff
@@ -0,0 +1,36 @@
+expert mode:
```
Comment (orltom) [inline charts/microgateway/tests/__snapshot__/configmap_test.yaml.snap:1]: **subjective opinion**
I prefer when file name start with test_

**Review by orltom — CHANGES_REQUESTED** [review]
GitHub Actions fails

File: charts/microgateway/tests/__snapshot__/configmap_test.yaml.snap (hunk 1 above)
Comment (libuweber) [reply charts/microgateway/tests/__snapshot__/configmap_test.yaml.snap:1]: I stuck to the naming conventions promoted by the unit testing tool we use. In this way, the tool is able to find the tests to run automatically. If we change the pattern, we would have to provide the test spec files to run in every invocation.

**Comment by libuweber** [comment]
> GitHub Actions fails

deplyoment temporarily disabled in the ci workflow, as the microgateway 2.0 images are not yet available on docker hub.

**Pull request merged**